src/sources/docker.rs

```diff
@@ -37,12 +38,26 @@ lazy_static! {
 
 type DockerEvent = shiplift::rep::Event;
 
-#[derive(Deserialize, Serialize, Debug, Clone, Default)]
-#[serde(deny_unknown_fields)]
+#[derive(Deserialize, Serialize, Debug, Clone)]
+#[serde(deny_unknown_fields, default)]
 pub struct DockerConfig {
     include_containers: Option<Vec<String>>,
     include_labels: Option<Vec<String>>,
     include_images: Option<Vec<String>>,
+    partial_event_marker_field: Option<Atom>,
```

Comment:
This goes back to my comment (https://github.com/timberio/vector/pull/1504#discussion_r369206256). Why would anyone want to do this? It seems like removing the `partial_event_marker_field` option is cleaner and simpler. And based on your comment above it seems like was the plan:

> Reimplement the docker source to eliminate partial_event_marker and extract an use a lower-level and more generic utility from the merge transform for in-code use.

Any reason you changed your mind or was that just an oversight?

Reply:
You're right, I said I wanted to remove the field, but not I don't. Actually, I just forgot about one of the use cases I did initially keep in mind. When migrating from other log aggregators, it's sometimes necessary to preserve the old behavior, even if it's broken. It might not make sense, but sometimes it's the only way to progress in the enterprise environment, so we should allow it.

Reply:
Sure, we can document it that way then 👍